pages/partials/studentGroupControls.ejs

```diff
@@ -0,0 +1,59 @@
+<% if (groupsize == 0) { %>
+  <% if (!permissions.student_authz_join && !permissions.student_authz_create) { %>
+    <p class="text-center">
+      This is a group homework. Please wait for the instructor to assign groups.
+   </p>
+  <% } else { %>
+<p class="text-center">
+   This is a group assessment.
+</p>
+<% if (typeof used_join_code !== 'undefined') { %>
+  <div class="alert alert-danger" role="alert">
+    Failed to join the group with join code <span class="badge badge-secondary"><%= used_join_code %></span>. It is already full or does not exist. Please try to join another one.
+  </div>
+<% } %>
+<% if (typeof invalidGroupName !== 'undefined') { %>
+  <div class="alert alert-danger" role="alert">
+    The group name is invalid. Alpha and numeric characters only. ([0-9] [a-z] [A-Z]), with maximum length of 30 characters.
+  </div>
+<% } %>
+<% if (typeof uniqueGroupName !== 'undefined') { %>
+  <div class="alert alert-danger" role="alert">
+    Failed to create the group. It is already taken. Please try another one.
+  </div>
+<% } %>
+<%- include('../shared/groupWorkInitial.ejs', {permissions: permissions}); %>
+<% } %>
+<% } else {%>
+  <%- include('../shared/groupWorkInfoContainer.ejs', {permissions: permissions, group_info: group_info}); %>
+  <% if (locals.assessment.require_honor_code) { %>
+    <div class="card card-secondary mb-4 test-class-honor-code">
+      <ul class="list-group list-group-flush">
+        <li class="list-group-item py-2">I certify that I am <%= user.name %> and our group is allowed to take this assessment.</li>
+        <li class="list-group-item py-2">We pledge on our honor that we will not give or receive any unauthorized assistance on this assessment and that all work will be our own.</li>
+      </ul>
+
+      <div class="card-footer text-center border-top-0 py-2">
+        <span class="form-check d-inline">
+          <input type="checkbox" class="form-check-input" id="certify-pledge">
+          <label class="form-check-label font-weight-bold" for="certify-pledge">I certify and pledge the above.</label>
+        </span>
+      </div>
+    </div>
+
+  <% } %>
+  <% if (start) { %>
+  <form id="confirm-form" name="confirm-form" method="POST" class="mt-4 d-flex justify-content-center">
+    <input type="hidden" name="__action" value="new_instance">
+    <input type="hidden" name="__csrf_token" value="<%= __csrf_token %>">
+    <button id="start-assessment" type="submit" class="btn btn-primary">Start assessment</button>
+  </form>
```

Comment:
The user can click "Start assessment" without having checked the "I certify..." box above. The "Start assessment" button should be disabled until the checkbox is checked. See the inline JS in `pages/studentAssessmentExam/studentAssessmentExam.ejs`.

<img width="590" alt="Screen Shot 2022-04-13 at 12 45 29" src="https://user-images.githubusercontent.com/1476544/163258251-5a1e512e-189f-4bce-b022-ef3857df52a9.png">

Reply:
Fixed in [7fd2129](https://github.com/PrairieLearn/PrairieLearn/pull/5290/commits/7fd21295df470998a82a0ff4a0864e8de376c073) (and removed a leftover, unused, copied script in [b4d3297](https://github.com/PrairieLearn/PrairieLearn/pull/5290/commits/b4d3297909aa1b57550eacc37e199bac4caf4d85) ) 

This issue did remind me of why I started trying things strangely in the first place. The way things are set up now, the majority of group homeworks will now require honor code, since for all assessments is `require_honor_code: boolean default true` and you hit that assessment homework page before the homework instance. So now it would have to be added explicitly in the `infoAssessment.json` that `"requireHonorCode":"false"` . I'm not sure if that would be desired or what to do about it.